text/0000-read-buf.md

```diff
@@ -0,0 +1,624 @@
+    ///
+    /// This includes the written-to portion.
+    #[inline]
+    pub fn initialized_mut(&mut self) -> &mut [u8] { ... }
+
+    /// Returns a mutable reference to the unwritten-to part of the buffer without ensuring that it has been fully
+    /// initialized.
+    ///
+    /// # Safety
+    ///
+    /// The caller must not de-initialize portions of the buffer that have already been initialized.
+    #[inline]
+    pub unsafe fn unwritten_mut(&mut self) -> &mut [MaybeUninit<u8>] { ... }
+
+    /// Returns a mutable reference to the unwritten-to part of the buffer, ensuring it is fully initialized.
+    ///
+    /// Since `ReadBuf` tracks the region of the buffer that has been initialized, this is effectively "free" after
+    /// the first use.
+    #[inline]
+    pub fn initialize_unwritten(&mut self) -> &mut [u8] { ... }
+
+    /// Returns a mutable reference to the first `n` bytes of the unwritten-to part of the buffer, ensuring it is
+    /// fully initialized.
+    ///
+    /// # Panics
+    ///
+    /// Panics if `self.remaining()` is less than `n`.
+    #[inline]
+    pub fn initialize_unwritten_to(&mut self, n: usize) -> &mut [u8] { ... }
+
+    /// Returns the number of bytes at the end of the slice that have not yet been written to.
+    #[inline]
+    pub fn remaining(&self) -> usize { ... }
+
+    /// Clears the buffer, resetting the written-to region to empty.
+    ///
+    /// The number of initialized bytes is not changed, and the contents of the buffer are not modified.
+    #[inline]
+    pub fn clear(&mut self) { ... }
+
+    /// Increases the size of the written-to region of the buffer.
+    ///
+    /// The number of initialized bytes is not changed.
+    ///
+    /// # Panics
+    ///
+    /// Panics if the written region of the buffer would become larger than the initialized region.
+    #[inline]
+    pub fn add_written(&mut self, n: usize) { ... }
+
+    /// Sets the size of the written-to region of the buffer.
+    ///
+    /// The number of initialized bytes is not changed.
+    ///
+    /// Note that this can be used to *shrink* the written region of the buffer in addition to growing it (for
+    /// example, by a `Read` implementation that compresses data in-place).
+    ///
+    /// # Panics
+    ///
+    /// Panics if the written region of the buffer would become larger than the intialized region.
+    #[inline]
+    pub fn set_written(&mut self, n: usize) { ... }
+
+    /// Asserts that the first `n` unwritten bytes of the buffer are initialized.
+    ///
+    /// `ReadBuf` assumes that bytes are never de-initialized, so this method does nothing when called with fewer
+    /// bytes than are already known to be initialized.
+    ///
+    /// # Safety
+    ///
+    /// The caller must ensure that `n` unwritten bytes of the buffer have already been initialized.
+    #[inline]
+    pub unsafe fn assume_initialized(&mut self, n: usize) { ... }
```

Comment:
`assume_init` would be more consistent with `MaybeUninit::assume_init`. But I guess at that point all "initialized" functions here should be called "init" instead. Maybe this is worth putting down as an open question to be decided later?